segmentation/utils/segmentation_utils.py

```diff
@@ -16,7 +16,110 @@
 import skimage.io as io
 
 
-from segmentation.utils import plot_utils, signal_extraction
+from segmentation.utils import data_utils, plot_utils, signal_extraction
+
+
+def compute_complete_expression_matrices(segmentation_labels, base_dir=None, tiff_dir=None,
+                                         img_sub_folder="TIFs", is_mibitiff=False, mibitiff_suffix=None,
+                                         points=None, batch_size=5):
+    """
+    This function takes the segmented data and computes the expression matrices batch-wise
+    while also validating inputs
+
+    Inputs:
+        base_dir (str): the "master" directory, contains all of the data including the input
+        tiff_dir (str): the name of the directory which contains the single_channel_inputs,
+            technically can be accessed from base_dir but easier to make its own separate arg
+        img_sub_folder (str): the name of the folder where the TIF images are located
+        points (list): a list of points we wish to analyze, if None will default to all points
+        is_mibitiff (bool): a flag to indicate whether or not the base images are MIBItiffs
+        mibitiff_suffix (str): if is_mibitiff is true, then needs to be specified to select
+            which points to load from mibitiff
+        segmentation_labels (xarray): an xarray with the segmented data
+        batch_size (int): how large we want each of the batches of points to be when computing,
+            adjust as necessary for speed and memory considerations
+
+    Returns:
+        combined_normalized_data (pandas): a DataFrame containing the data with the size_norm transformation
+        combined_transformed_data (pandas): a DataFrame containing the data with the arcsinh transformation
+    """
+
+    if not base_dir:
+        raise ValueError("base_dir not specified")
+
+    if not tiff_dir:
+        raise ValueError("tiff_dir not specified")
+
+    if not img_sub_folder:
+        raise ValueError("img_sub_folder not specified")
+
+    # if no points are specified, then load all the points
+    if not points:
```

Comment:
I like `if points is None`, I think it's clearer to understand

Reply:
👍